### PR TITLE
Report all errors as messages not as Exceptions

### LINKED ISSUE #4912: lakefs_export: Report errors as messages not exceptions
lakefs_export reports errors by raising Python Exceptions.  This is hard for
its users to read, at least for `'@arielshaqed' in users`.  Report errors in
the regular way: write to standard error and then exit with a nonzero status
code.

Discovered while working on #4904, #4841.

### PR DESCRIPTION
**Pull request by arielshaqed**
This improves the UX and makes thing more scriptable and more UN*X-like.

Tested by manually running these failing commands:
```sh
❯ ./lakefs_export.py --branch asdf --commit_id sdf abc def
Cannot set both branch and commit

❯ ./lakefs_export.py abc def
Must set one of branch, commit

❯ ./lakefs_export.py --prev_commit_id asdf --commit_id sdf abc def
Cannot export diff between two commits.
```

Fixes #4912.

### CODE REVIEW

File: deployments/tools/export/lakefs_export.py [hunk 1]
```diff
@@ -78,11 +82,13 @@ def main():
     elif not has_branch and has_commit:
         source += args.commit_id + "/"
         reference = args.commit_id
-    else:
-        raise Exception("Should assign branch or commit, but not both.")
+    elif has_branch:            # and has_commit
+        error("Cannot set both branch and commit_id")
+    else:                       # not has_branch and not has_commit
+         error("Must set one of branch, commit_id")
```
Comment (johnnyaug) [inline deployments/tools/export/lakefs_export.py:88]: strange indentation here

**Review by johnnyaug — APPROVED** [review]

File: deployments/tools/export/lakefs_export.py (hunk 1 above)
Comment (arielshaqed) [reply deployments/tools/export/lakefs_export.py]: Thanks!  Turns out we never flake8 this.  Opening a _new_ bug to do that, but for now fixed _everything_ flake8 had to complain about.  So as punishment PTAL :-/

**Review by johnnyaug — APPROVED** [review]
LGTM

**Comment by arielshaqed** [comment]
Thanks muchly!

**Pull request merged**